src/components/common/PeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -364,55 +372,50 @@ export default class PeopleAndRoles extends Mixins(CommonMixin) {
     // get org/person to undo
     const person = tempList[index]
 
-    switch (person.action) {
-      case ActionTypes.ADDED:
-        // splice out the person
-        tempList.splice(index, 1)
-        break
-
-      case ActionTypes.EDITED:
-      case ActionTypes.REMOVED:
-        // get ID of person to undo
-        const id = person?.officer?.id
-
-        // get a copy of original person from original IA
-        const thisPerson = (id !== undefined) && cloneDeep(this.originalParties.find(x => x.officer.id === id))
-
-        // safety check
-        if (!thisPerson) {
-          // eslint-disable-next-line no-console
-          console.log('Failed to find original person with id =', id)
-          return
-        }
-
-        // check if original person had CP role
-        const hadCp = thisPerson.roles.some(role => role.roleType === RoleTypes.COMPLETING_PARTY)
-
-        // check if an other person has CP role right now
-        // NB: we will update this record right in the temp list - no need to splice
-        const otherPerson = this.getCompletingParty(tempList)
-
-        // check if we would restore the original CP
-        if (hadCp && otherPerson && (thisPerson.officer.id !== otherPerson.officer.id)) {
-          // prompt user for confirmation
-          await this.confirmReassignCp().then(confirm => {
-            if (confirm) {
-              // remove the other person's CP role and their email
-              otherPerson.roles = otherPerson.roles.filter(r => r.roleType !== RoleTypes.COMPLETING_PARTY)
-              delete otherPerson.officer.email
-              otherPerson.action = this.computeAction(otherPerson)
-            } else {
-              // remove this person's CP role and their email
-              thisPerson.roles = thisPerson.roles.filter(r => r.roleType !== RoleTypes.COMPLETING_PARTY)
-              delete thisPerson.officer.email
-              thisPerson.action = this.computeAction(thisPerson)
-            }
-          })
-        }
-
-        // splice in the original person
-        tempList.splice(index, 1, thisPerson)
-        break
+    if (person.action.includes(ActionTypes.ADDED)) {
+      // splice out the person
+      tempList.splice(index, 1)
+    } else {
+      // get ID of person to undo
+      const id = person?.officer?.id
+
+      // get a copy of original person from original IA
+      const thisPerson = (id !== undefined) && cloneDeep(this.originalParties.find(x => x.officer.id === id))
+
+      // safety check
+      if (!thisPerson) {
+        // eslint-disable-next-line no-console
+        console.log('Failed to find original person with id =', id)
```

Comment:
As this should never happen, it means we screwed up somewhere. Therefore, I recommend changing this to `console.error()`.

Reply:
Sounds good, will do!